url2892692420_mikaello_inf3110_oblig1_tgz-pJ8-inf3110_oblig1_Inf3110_oblig1J8/src/util/Grid.java

```diff
@@ -0,0 +1,176 @@
+package util;
+
+import expressions.Numbers;
+import interfaces.Handler;
+import java.util.HashMap;
+import log.Log;
+import statements.Size;
+
+/**
+ * This class prints the grid. It contains data about how big it is, 
+ * but no other info is stored here.
+ * @author mikaello
+ */
+public class Grid implements Handler {
+    /** Size of this map */
+    private Size sizeGlobal = new Size(new Numbers(0), new Numbers(0));
+    
+    public Grid() {
+    }
+    
+    public Grid(Size sizeIns) {
+        Log.writeLogLine(sizeIns.toString());
+        setSizeGlobal(sizeIns);
+    }
+    
+    
+    @Override
+    public void interpret() {
+    }
+    
+    /**
+     * @see printGrid(int, int, String)
+     */
+    public void printGrid() {
+        printGrid(new Position(new Numbers(1), new Numbers(1)));
+    }
+    
+    /**
+     * @see printGrid(int, int, String)
+     */
+    public void printGrid(Position p) {
+        HashMap<Position, Direction> pList = new HashMap<Position, Direction>();
+        
+        // Must make new Position to get a Direction to be standby 
+        // without altering the original Position p
+        pList.put(new Position(new Numbers(p.getXCord()), 
+                new Numbers(p.getYCord())), p.getDirection());
+        
+        printGrid(pList, p);
+    }
+    
+    public void printGrid(Numbers x, Numbers y, Direction symbol) {      
+        printGrid(new Position(x, y, symbol));
+    }
+    
+    /**
+     * Makes a grid of dots. At the point (x,y) it will print the
+     * symbol argument instead of dots.
+     *
+     * @param x x coordinate (starts from 1)
+     * @param y y coordinate (starts from 1)
+     * @param symbol symbol of robot to be filled
+     */
+    public void printGrid(HashMap<Position, Direction> positionMap, Position lastPos) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 134-151 into a separate method.